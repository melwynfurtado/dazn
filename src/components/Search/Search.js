import React from 'react'
import PropTypes from 'prop-types'
import './Search.scss'

const Search = ({ query, handleOnChange }) => (
  <input className="search-field" type="text" name="query" value={query} onChange={handleOnChange} />
)

Search.propTypes = {
  query: PropTypes.string,
  handleOnChange: PropTypes.func.isRequired,
}

export default Search