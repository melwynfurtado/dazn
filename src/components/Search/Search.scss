.search-field {
  width: 90%;
  height: 30px;
  border-radius: 5px;
  border: 1px solid lightgrey;
  padding: 3px 10px;
}