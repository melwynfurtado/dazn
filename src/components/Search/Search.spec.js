import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Search from './Search'

describe('Search Component', () => {
  let handleOnChangeMock
  let Component

  beforeEach(() => {
    handleOnChangeMock = jest.fn()
    Component = <Search handleOnChange={handleOnChangeMock} />
  })

  it('renders correctly', () => {
    const tree = renderer
                  .create(Component)
                  .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('calls handleOnChange on query change', () => {
    const wrapper = shallow(Component)
    wrapper.simulate('change')
    expect(handleOnChangeMock).toHaveBeenCalled()
  })
})