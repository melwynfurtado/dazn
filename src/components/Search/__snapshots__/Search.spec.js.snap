// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component renders correctly 1`] = `
<input
  className="search-field"
  name="query"
  onChange={[MockFunction]}
  type="text"
/>
`;
